miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/RealMiniApp.kt

```diff
@@ -30,6 +30,18 @@ internal class RealMiniApp(
     override fun setCustomPermissions(miniAppCustomPermission: MiniAppCustomPermission) =
         miniAppCustomPermissionCache.storePermissions(miniAppCustomPermission)
 
+    @Suppress("FunctionMaxLength")
+    override fun listDownloadedWithCustomPermissions(): List<Pair<MiniAppInfo, MiniAppCustomPermission>> {
+        val result: ArrayList<Pair<MiniAppInfo, MiniAppCustomPermission>> = arrayListOf()
+        val downloadedMiniApps = miniAppDownloader.getDownloadedMiniAppList()
+
+        downloadedMiniApps.forEach {
+            result.add(Pair(it, miniAppCustomPermissionCache.readPermissions(it.id)))
+        }
+
+        return result
```

Comment:
You could use `map` and avoid needing to create a temporary list:

```suggestion
        return miniAppDownloader.getDownloadedMiniAppList().map {
            Pair(it, miniAppCustomPermissionCache.readPermissions(it.id))
        }
```

Reply:
Thanks, I have updated it.